core/commands/checkout.py

```diff
@@ -51,21 +46,24 @@ class GsCheckoutNewBranchCommand(WindowCommand, GitCommand):
     Prompt the user for a new branch name, create it, and check it out.
     """
 
-    def run(self):
-        sublime.set_timeout_async(self.run_async)
+    def run(self, base_branch=None):
+        sublime.set_timeout_async(lambda: self.run_async(base_branch))
 
-    def run_async(self, name=""):
-        self.window.show_input_panel(NEW_BRANCH_PROMPT, name, self.on_done, None, None)
+    def run_async(self, base_branch=None):
+        self.base_branch = base_branch
+        self.window.show_input_panel(NEW_BRANCH_PROMPT, base_branch or "", self.on_done, None, None)
 
     def on_done(self, branch_name):
-        pattern = r"^(?!\.|.*\.\..*|.*@.*|\/)[a-zA-Z0-9\-\_\/\.\u263a-\U0001f645]+(?<!\.lock)(?<!\/)(?<!\.)$"
-        match = re.match(pattern, branch_name)
+        match = _is_valid_branch_name.match(branch_name)
         if not match:
-            sublime.error_message("`{}` is a invalid branch name.\nRead more on $(man git-check-ref-format)".format(branch_name))
+            sublime.error_message(NEW_BRANCH_INVALID.format(branch_name))
             sublime.set_timeout_async(self.run_async(branch_name))
             return None
 
-        self.git("checkout", "-b", branch_name)
+        self.git(
+            "checkout", "-b",
+            branch_name,
+            self.base_branch if self.base_branch else None)
```

Comment:
It is useful to specific the starting point in some cases. The default value of `base_branch` is None. It would not produce any side effects in the normal use case.